proto/regen/ecocredit/marketplace/v1/tx.proto

```diff
@@ -195,6 +183,37 @@ message MsgBuyResponse {
   repeated uint64 buy_order_ids = 1;
 }
 
+// MsgBuyDirect is the Msg/BuyDirect request type.
+message MsgBuyDirect {
+
+  // buyer is the address of the credit buyer.
+  string buyer = 1;
+
+  // sell_order_id is the sell order ID against which the buyer is trying
+  // to buy.
+  uint64 sell_order_id = 2;
+
+  // quantity is the quantity of credits to buy.
+  string quantity = 3;
+
+  // bid_price is the price the buyer is willing to pay per credit.
+  cosmos.base.v1beta1.Coin bid_price = 4;
+
+  // disable_auto_retire allows auto-retirement to be disabled. If it is set
+  // to true the credits will not auto-retire and can be resold assuming that
+  // the corresponding sell order has auto-retirement disabled. If the sell
+  // order hasn't disabled auto-retirement and the buy order tries to disable
+  // it, that buy order will fail.
+  bool disable_auto_retire = 5;
+
+  // retirement_location is the optional retirement location for the credits
+  // which will be used only if disable_auto_retire is false.
+  string retirement_location = 6;
+}
```

Comment:
We use `repeated Order` in `Buy` to support multiple buy orders. I think we should preserve that here. Imagine a user browsing the marketplace, selecting multiple sell orders (i.e. "add to cart"), and then purchasing.

Reply:
added: https://github.com/regen-network/regen-ledger/pull/967/commits/fe36dd2c3c8ef771c5df022bcc90f181cbe82cf2